src/_posts/2019-02-25-dynamodb-explained-part-1.md

```diff
@@ -0,0 +1,62 @@
+---
+layout: post
+asset-type: post
+name: 2019-02-25-dynamodb-explained-part-1
+title: DynamoDB explained - Part 1
+date: 2019-02-25 10:15:00 +00:00
+author: Bartomeu Galmés
+description: Deep diving on DynamoDB
+image: 
+    src: /assets/custom/img/blog/2019-02-25-dynamodb-explained-part-1/00-cloud.jpg
+tags: 
+- dynamodb
+- amazon web services
+- aws
+- databases
+- cloud
+
+---
+
+
+DynamoDB has been a bit of a buzzword recently, so after years working on relational and document databases I decided to give it a try. A lot of the assumptions I had made about DynamoDb before working with it turned out to be false, so I am hoping that this series of blog posts will help you decide if DynamoDB is appropriate for your next project.
+
+In this series of posts we will dive into DynamoDB and discover some of its quirks, which can differ a little from other databases. In this first post we will talk about what DynamoDB is and how it's approach to primary keys works. In the next post we will examine one of the most confusing aspects for people coming to DynamoDB from other databases - data retrieval - which works a little differently. We will see that, with a little help from indexes, we can achieve the most common use cases. The third and final part of this series will be focused on availability and scalability which are two big advantages of DynamoDB as AWS handles them for us.
+
+Before getting started let's see where DynamoDB lives inside the database categories.
+
+![alt text]({{site.baseurl}}/assets/custom/img/blog/2019-02-25-dynamodb-explained-part-1/01-dynamodb-db-tree.jpg "DynamoDB inside the DB family")
+
+# What is DynamoDB?
+
+DynamoDB is a NoSQL, highly-available, low latency, scalable database that supports both key-value and document store data models. It follows a serverless approach, meaning that it is offered as a service, so the infrastructure is fully managed by AWS and works out-of-the-box with little configuration. This frees the user from managing instances, applying security patches and other infrastructure-related tasks. In addition, pricing can be very reasonable as it follows a pay-per-use model instead of charging for pre-provisioned computing resources.
+
+As there is no instance the database concept, existent on most relational databases and also in some NoSQL like MongoDB, disappears and all is done on tables. Moreover, tables are schemaless meaning that the objects stored are not required to have the same attributes, providing more flexibility to the user.
```

Comment:
I think there is something missing in "As there is no instance the database concept"

Reply:
Oh I get the meaning now, I think the explanatory sentence after the comma is too long and makes the paragraph difficult to read/understand